File: app/src/main/java/com/scatl/uestcbbs/module/message/adapter/ReplyMeMsgAdapter.java
package com.scatl.uestcbbs.module.message.adapter;

import android.view.View;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.scatl.uestcbbs.R;
import com.scatl.uestcbbs.entity.ReplyMeMsgBean;
import com.scatl.uestcbbs.manager.MessageManager;
import com.scatl.uestcbbs.util.TimeUtil;

/**
 * author: sca_tl
 * description:
 * date: 2020/1/29 15:27
 */
public class ReplyMeMsgAdapter extends BaseQuickAdapter<ReplyMeMsgBean.BodyBean.DataBean, BaseViewHolder> {
    public ReplyMeMsgAdapter(int layoutResId) {
        super(layoutResId);
    }

    @Override
    protected void convert(BaseViewHolder helper, ReplyMeMsgBean.BodyBean.DataBean item) {
        helper.setText(R.id.user_name, item.user_name)
                .setText(R.id.board_name, "来自板块:" + item.board_name)
                .setText(R.id.reply_content, (item.reply_content == null ? "" : item.reply_content)
                        .replace("\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n", "\n")
                        .replace("\r\n", "")
                )
                .setText(R.id.subject_title, item.topic_subject)
                .setText(R.id.subject_content,  (item.topic_content == null ? "" : item.topic_content)
                        .replace("\r\n", "")
                )
                .setText(R.id.reply_date, TimeUtil.formatTime(item.replied_date, R.string.post_time1, mContext))
                .addOnClickListener(R.id.user_icon)
                .addOnClickListener(R.id.subject_detail)
                .addOnClickListener(R.id.reply_btn)
                .addOnClickListener(R.id.board_name);
        Glide.with(mContext).load(item.icon).into((ImageView) helper.getView(R.id.user_icon));

        //显示未读标志
        helper.getView(R.id.new_msg_img).setVisibility(
                helper.getLayoutPosition() < MessageManager.Companion.getINSTANCE().getReplyUnreadCount() ?
                        View.VISIBLE : View.GONE);
    }
}

File: app/src/main/java/com/scatl/uestcbbs/module/message/adapter/ReplyMeMsgAdapter.kt
package com.scatl.uestcbbs.module.message.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.scatl.uestcbbs.databinding.ItemReplyMeMsgBinding
import com.scatl.uestcbbs.entity.ReplyMeMsgBean
import com.scatl.uestcbbs.helper.PreloadAdapter
import com.scatl.uestcbbs.helper.ViewBindingHolder
import com.scatl.uestcbbs.R
import com.scatl.uestcbbs.manager.MessageManager
import com.scatl.uestcbbs.util.TimeUtil
import com.scatl.uestcbbs.util.load

/**
 * Created by sca_tl at 2023/6/12 14:49
 */
class ReplyMeMsgAdapter: PreloadAdapter<ReplyMeMsgBean.BodyBean.DataBean, ItemReplyMeMsgBinding>() {

    override fun getViewBinding(parent: ViewGroup): ItemReplyMeMsgBinding {
        return ItemReplyMeMsgBinding.inflate(LayoutInflater.from(context), parent, false)
    }

    override fun onBindViewHolder(holder: ViewBindingHolder<ItemReplyMeMsgBinding>, position: Int, item: ReplyMeMsgBean.BodyBean.DataBean?) {
        super.onBindViewHolder(holder, position, item)
        if (item == null) {
            return
        }

        holder.binding.userName.text = item.user_name
        holder.binding.boardName.text = "来自板块:${item.board_name}"
        holder.binding.replyDate.text = TimeUtil.formatTime(item.replied_date, R.string.post_time1, context)
        holder.binding.userIcon.load(item.icon)
        holder.binding.subjectTitle.apply {
            text = if (item.topic_subject.isNullOrEmpty()) {
                ""
            } else {
                item.topic_subject.replace("\r\n", "")
            }
            visibility = if (item.topic_subject.isNullOrEmpty()) View.GONE else View.VISIBLE
        }
        holder.binding.replyContent.apply {
            text = if (item.reply_content.isNullOrEmpty()) {
                ""
            } else {
                item.reply_content
                    .replace("\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n\r\n", "\n")
                    .replace("\r\n", "")
            }
            visibility = if (item.reply_content.isNullOrEmpty()) View.GONE else View.VISIBLE
        }

        //显示未读标志
        holder.binding.newMsgImg.visibility =
            if (holder.layoutPosition < MessageManager.INSTANCE.replyUnreadCount) View.VISIBLE else View.GONE
    }
}